packages/expo-modules-core/android/src/main/java/expo/modules/adapters/react/services/UIManagerModuleWrapper.java

```diff
@@ -54,11 +59,32 @@ public List<Class> getExportedInterfaces() {
     );
   }
 
+  private void addToUIManager(final UIBlockInterface block) {
+    if (BuildConfig.IS_NEW_ARCHITECTURE_ENABLED) {
+      com.facebook.react.bridge.UIManager uiManager = UIManagerHelper.getUIManager(getContext(), UIManagerType.FABRIC);
+      assert uiManager != null;
+      ((FabricUIManager) uiManager).addUIBlock(block);
+    } else {
+      UIManagerModule uiManager = getContext().getNativeModule(UIManagerModule.class);
+      assert uiManager != null;
```

Comment:
```suggestion
      Assertions.assertNotNull(uiManager);
```

the assert in android java vm is not deterministic